app/src/main/java/ru/firstSet/kotlinOne/worker/RepositorySP.kt

```diff
@@ -0,0 +1,49 @@
+package ru.firstSet.kotlinOne.worker
+
+import android.content.Context
+import android.content.SharedPreferences
+import android.util.Log
+import java.text.SimpleDateFormat
+import java.util.*
+
+class RepositorySP(context: Context) {
+
+    private var movieSharedPreferences: SharedPreferences = context.getSharedPreferences(
+        SHARED_PREFERENCES, Context.MODE_PRIVATE
+    )
+    private lateinit var editorSharedPreferences: SharedPreferences.Editor
+
+    fun saveCurrentDate() {
+        val date: String = RepositorySP.sdf.format(
+            Calendar.getInstance().time
+        )
+        editorSharedPreferences = movieSharedPreferences.edit()
+        editorSharedPreferences.putString(KEY_DATE_UPDATE, date)
+        editorSharedPreferences.apply()
+        Log.v("saveCurrentDate", "${date}")
+    }
+
+    fun getValue(): String? {
+        return movieSharedPreferences.getString(KEY_DATE_UPDATE, "")
+    }
+
+    fun compareSavedDate(): Long {
```

Comment:
@kamar19 This logic is not very good solution. Please check how to do it via Room automatically https://developer.android.com/training/data-storage/room/async-queries **Write observable queries** topic you need.